<h3 style="margin:0px">Class: org.apache.commons.math4.RetryRunnerTest (2 methods) </h3><br><pre style="padding:0"><code class="html"><div id="tags"><button id="8"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('8')" data-toggle="tooltip" title="Is not categorized by any of the available tags"><kbd id="tag-8"class="label-info"style="display: inline-block;font-size:7pt" >Unclassified&nbsp;(1)</kbd></button>&nbsp;<button id="7"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('7')" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution"><kbd id="tag-7"class="label-info"style="display: inline-block;font-size:7pt" >ExceptionVerifier&nbsp;(1)</kbd></button>&nbsp;</div></code></pre><center><button onclick="showBlocks()" style="margin:6px">Update filter <span class="glyphicon glyphicon-refresh" aria-hidden="true"></span> </button></center><br>
<pre class="type-7 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
"></span><br>
/** 
 * Shows that an always failing test will fail even if it is retried.
 */
@Test(expected=MathIllegalStateException.class) @Retry public void testRetryFailAlways(){
  throw new MathIllegalStateException();
}

</code></pre>

<pre class="type-8 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Is not categorized by any of the available tags">Unclassified</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Is not categorized by any of the available tags
"></span><br>
/** 
 * Shows that a test that sometimes fail might succeed if it is retried.
 * In this case the high number of retries makes it quite unlikely that
 * the exception will be thrown by all of the calls.
 */
@Test @Retry(100) public void testRetryFailSometimes(){
  if (rng.nextBoolean()) {
    throw new MathIllegalStateException();
  }
}

</code></pre>

<script>$(document).ready(function() {
  $('pre code').each(function(i, block) {
    hljs.highlightBlock(block);
  });
});</script>
